tests/test_utils.py

```diff
@@ -290,13 +292,211 @@ def test_random_interferometer(self):
         self.assertAllAlmostEqual(U @ U.conj().T, np.identity(self.M), delta=self.tol)
 
 
+class ExtractChannel(ExtractChannelTest):
+    num_subsystems = 3
+
+    def setUp(self):
+        super().setUp()
+
+    def test_extract_unitary_1_mode(self):
+        num_subsystems = 1
+
+        eng_sf, q_sf = sf.Engine(num_subsystems=num_subsystems)
+        eng_extract, q_extract = sf.Engine(num_subsystems=num_subsystems)
+
+        cutoff_dim = 10
+
+        S = sf.ops.Sgate(0.4, -1.2)
+        D = sf.ops.Dgate(2, 0.9)
+        K = sf.ops.Kgate(-1.5)
+
+        initial_state = np.random.rand(cutoff_dim) + 1j*np.random.rand(cutoff_dim) # not a state but it doesn't matter
+
+        with eng_sf:
+            sf.ops.Ket(initial_state) | q_sf
+            S | q_sf
+            D | q_sf
+            K | q_sf
+
+        with eng_extract:
+            S | q_extract
+            D | q_extract
+            K | q_extract
+
+        matrix_extract = extract_unitary(eng_extract, cutoff_dim=cutoff_dim)
+        matrix_extract_tf_backend = extract_unitary(eng_extract, cutoff_dim=cutoff_dim, backend='tf')
+        with tf.Session() as sess:
+            sess.run(tf.global_variables_initializer())
+            in_state = tf.constant(initial_state.reshape([-1]), dtype=tf.complex64)
+            final_state_tf_backend = sess.run(tf.einsum('ab,b', matrix_extract_tf_backend, in_state))
+        final_state_sf = eng_sf.run("fock", cutoff_dim=cutoff_dim).ket().reshape([-1])
+        final_state_extract = matrix_extract.dot(initial_state)
+        final_state_sf = eng_sf.run("fock", cutoff_dim=cutoff_dim).ket()
+
+        self.assertAllAlmostEqual(final_state_extract, final_state_sf, delta=defaults.TOLERANCE)
+        self.assertAllAlmostEqual(final_state_tf_backend, final_state_sf, delta=defaults.TOLERANCE)
```

Comment:
Is there a reason here why you are testing the circuit twice, once with the tf backend and once with the Fock backend? The way we have the tests set up, the backend should be dynamically chosen based on an environment variable (`BACKEND=fock` or `BACKEND=tf`) in `defaults.py`, allowing us to run the same tests with different backends

Reply:
No, no reason, I just had not understood that your tests can do that switch automatically 😅